src/libos/src/fs/procfs/stat_inode.rs

```diff
@@ -0,0 +1,20 @@
+use super::*;
+use crate::fs::hostfs::IntoFsError;
+use std::untrusted::fs;
+
+/// An untrusted implementation of stat, it return the information from host OS's "/proc/stat"
+pub struct StatINode;
+
+impl StatINode {
+    pub fn new() -> Arc<dyn INode> {
+        Arc::new(File::new(Self))
+    }
+}
+
+impl ProcINode for StatINode {
+    fn generate_data_in_bytes(&self) -> vfs::Result<Vec<u8>> {
+        Ok(fs::read_to_string("/proc/stat")
+            .map_err(|e| e.into_fs_error())?
```

Comment:
Can we do any simple check to confirm it is really a /proc/stat file?

Reply:
I think the check here is not trivial,  you have to check the file if it is in some predefined format, maybe some regex match tools can be used here. BTW, I think the effort is more than just implement a safe one.